### ISSUE TITLE
关于训练

### ISSUE THREAD

**Issue by guggugg** [yeyupiaoling/VoiceprintRecognition-Pytorch#3]
你好，我用ST-CMDS-20170001_1-OS这个数据集训练了5个epoch了，准确率一直为0，这是正常的吗？
[2021-09-24 14:11:23.609168] Train epoch 5, batch: 3500/5322, loss: 17.497463, accuracy: 0.000000, eta: 11:55:58
[2021-09-24 14:11:59.211179] Train epoch 5, batch: 3600/5322, loss: 21.458420, accuracy: 0.000000, eta: 11:43:53
[2021-09-24 14:12:34.664167] Train epoch 5, batch: 3700/5322, loss: 18.574211, accuracy: 0.000000, eta: 11:47:35
[2021-09-24 14:13:09.979111] Train epoch 5, batch: 3800/5322, loss: 19.353443, accuracy: 0.000000, eta: 11:51:12
[2021-09-24 14:13:45.611116] Train epoch 5, batch: 3900/5322, loss: 19.091064, accuracy: 0.000000, eta: 11:42:59
[2021-09-24 14:14:20.909075] Train epoch 5, batch: 4000/5322, loss: 24.289619, accuracy: 0.000000, eta: 11:58:24

**Comment by yeyupiaoling** [comment]
这个不应该，你设置多少个类别

**Comment by guggugg** [comment]
喔 糟了 我忘记改了....但是我train_list.txt是乱七八糟的顺序，不知道有多少类
E:/SpeechProcessing/datasets/ST-CMDS-20170001_1-OS/20170001P00219I0094.wav	671
E:/SpeechProcessing/datasets/ST-CMDS-20170001_1-OS/20170001P00116A0016.wav	700
E:/SpeechProcessing/datasets/ST-CMDS-20170001_1-OS/20170001P00359I0054.wav	685
E:/SpeechProcessing/datasets/ST-CMDS-20170001_1-OS/20170001P00300I0108.wav	74
E:/SpeechProcessing/datasets/ST-CMDS-20170001_1-OS/20170001P00317A0065.wav	657
E:/SpeechProcessing/datasets/ST-CMDS-20170001_1-OS/20170001P00225I0031.wav	372
E:/SpeechProcessing/datasets/ST-CMDS-20170001_1-OS/20170001P00186I0067.wav	715

**Comment by yeyupiaoling** [comment]
那你得看一下这个有多少类，然后修改训练代码中的类别数量，另外你的数据集生成正确吗？

**Comment by guggugg** [comment]
因为这个数据集没有json文件，所以我用的您另一个项目的这个文件生成的https://github.com/yeyupiaoling/VoiceprintRecognition-Tensorflow/blob/master/create_data.py生成的train_list.txt，共42570条数据，应该也可以在这个项目训练吧？

我刚把类别改成了855，重新训练了，期待不会有错，感谢您指出我的错误。

**Comment by yeyupiaoling** [comment]
只要数据准确，应该是没有问题的

**Comment by guggugg** [comment]
您好，还想问一下，关于这个项目的声纹识别模型有相关论文什么的么？

**Comment by yeyupiaoling** [comment]
参考了这个：https://arxiv.org/abs/1801.07698

**Comment by guggugg** [comment]
您好，我更改类别之后重新训练accuracy仍一直是0，同样的train_list.txt在tensorflow版本运行accuracy有变化耶，请问可能是哪方面的原因呢？

**Comment by yeyupiaoling** [comment]
可能是数据有问题，或者训练20个epoch在看看

**Comment by guggugg** [comment]
您好，我训练了50epoch之后效果很好了，感谢您的分享。还有一些小问题可以请教一下么

1. input_shape为啥是[1, 257, 257]呢？音频不是一维的吗。
2. 我看一般加载的数据都是[1,32000]这样的，您的代码在哪里处理成[1, 257, 257]这样维度的呢

**Comment by yeyupiaoling** [comment]
你自己看：

https://github.com/yeyupiaoling/VoiceprintRecognition-Pytorch/blob/61b91670cf4d448de9afae562f4648a8a4048a8d/utils/reader.py#L7-L34
